opensearch_acls.go

```diff
@@ -0,0 +1,129 @@
+package aiven
+
+import "context"
+
+type (
+	// OpenSearchACLsHandler Aiven go-client handler for OpenSearch ACLs
+	OpenSearchACLsHandler struct {
+		client *Client
+	}
+
+	// OpenSearchACLRequest Aiven API request
+	// https://api.aiven.io/v1/project/<project>/service/<service_name>/opensearch/acl
+	OpenSearchACLRequest struct {
+		OpenSearchACLConfig OpenSearchACLConfig `json:"opensearch_acl_config"`
+	}
+
+	// OpenSearchACLResponse Aiven API response
+	// https://api.aiven.io/v1/project/<project>/service/<service_name>/opensearch/acl
+	OpenSearchACLResponse struct {
+		APIResponse
+		OpenSearchACLConfig OpenSearchACLConfig `json:"opensearch_acl_config"`
+	}
+
+	// OpenSearchACLConfig represents a configuration for OpenSearch ACLs
+	OpenSearchACLConfig struct {
+		ACLs        []OpenSearchACL `json:"acls"`
+		Enabled     bool            `json:"enabled"`
+		ExtendedAcl bool            `json:"extendedAcl"`
+	}
+
+	// OpenSearchACL represents a OpenSearch ACLs entry
+	OpenSearchACL struct {
+		Rules    []OpenSearchACLRule `json:"rules"`
+		Username string              `json:"username"`
+	}
+
+	// OpenSearchACLRule represents a OpenSearch ACLs Rule entry
+	OpenSearchACLRule struct {
+		Index      string `json:"index"`
+		Permission string `json:"permission"`
+	}
+)
+
+// Update updates OpenSearch ACL config
+func (h *OpenSearchACLsHandler) Update(ctx context.Context, project, service string, req OpenSearchACLRequest) (*OpenSearchACLResponse, error) {
+	path := buildPath("project", project, "service", service, "opensearch", "acl")
+	bts, err := h.client.doPutRequest(ctx, path, req)
+	if err != nil {
+		return nil, err
+	}
+
+	var r OpenSearchACLResponse
+	errR := checkAPIResponse(bts, &r)
+
+	return &r, errR
```

Comment:
do that instead:

```go
	var r OpenSearchACLResponse
	return &r, checkAPIResponse(bts, &r)
```

and everywhere else where applicable

Reply:
I only did the ones in `opensearch_acls.go` and `elasticsearch_acls.go`. Felt making this change everywhere belongs in a separate PR.

Reply:
I meant within the scope of this PR :D